content/en/docs/concepts/cluster-administration/flow-control.md

```diff
@@ -782,6 +782,106 @@ APF adds the following two headers to each HTTP response message.
 - `X-Kubernetes-PF-PriorityLevel-UID` holds the UID of the
   PriorityLevelConfiguration object associated with that FlowSchema.
 
+## Good practices for using API Priority and Fairness
+When a given priority level exceeds its permitted concurrency, requests can 
+experience increased latency or be dropped with an HTTP 429 (Too Many Requests) 
+error. To prevent these side effects of APF, you can modify your workload or 
+tweak your APF settings to ensure there are sufficient seats available to serve
+your requests.
+
+To detect whether requests are being rejected due to APF, check the following
+metrics:
+- apiserver_flowcontrol_rejected_requests_total: the total number of requests 
+rejected per FlowSchema and PriorityLevelConfiguration.
+- apiserver_flowcontrol_current_inqueue_requests: the current number of requests
+queued per FlowSchema and PriorityLevelConfiguration.
+- apiserver_flowcontrol_request_wait_duration_seconds: the latency added to
+requests waiting in queues.
+- apiserver_flowcontrol_priority_level_seat_utilization: the seat utilization 
+per PriorityLevelConfiguration.
+
+### Workload modifications {#good-practice-workload-modifications}
+To prevent requests from queuing and adding latency or being dropped due to APF, 
+you can optimize your requests by:
+
+- Reducing the rate at which requests are executed. A fewer number of requests 
+over a fixed period will result in a fewer number of seats being needed at a 
+given time.
+- Avoid issuing a large number of expensive requests concurrently. Requests can 
+be optimized to use fewer seats or have lower latency so that these requests 
+hold those seats for a shorter duration. List requests can occupy more than 1 
+seat depending on the number of objects fetched during the request. Restricting 
+the number of objects retrieved in a list request (by using pagination or 
+limiting the request to a particular namespace) will use less total seats over a
+shorter period. Furthermore, replacing list requests with watch requests will 
+require lower total concurrency shares as watch requests only occupy 1 seat 
+during its initial burst of notifications.
```

Comment:
This will not be true anymore with watch-lists

Reply:
Can you point me to where I can read about the seat calculation used for watch-lists (or link the relevant PR)? 

For my understanding, how do WatchLists differ from watch requests which do not set a resourceVersion but set allowWatchBookmarks to true?
```
kubectl get --raw "/api/v1/namespaces/default/pods?watch=true&allowWatchBookmarks=true"
```
Without the feature enabled, we get init events of type ADDED and a BOOKMARK event, does WatchList provide a guarantee on the BOOKMARK event being sent?

Reply:
> Can you point me to where I can read about the seat calculation used for watch-lists (or link the relevant PR)?

https://github.com/kubernetes/kubernetes/pull/118933

> Without the feature enabled, we get init events of type ADDED and a BOOKMARK event, does WatchList provide a guarantee on the BOOKMARK event being sent?

It gives guarantees about the point in time when bookmark is sent and allows us to use that.
You can read much more in: https://github.com/kubernetes/enhancements/tree/master/keps/sig-api-machinery/3157-watch-list